File: 0016-3sum-closest/0016-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {

        int closest = nums[0] + nums[1] + nums[2];
        Arrays.sort(nums);

        for(int i=0;i<nums.length -2;i++){
            int left = i+1;
            int right = nums.length -1;

            while(left < right){
                int currentSum = nums[i] + nums[left] + nums[right];

                if(Math.abs(target - currentSum) < Math.abs(target - closest)) closest = currentSum;


                if(currentSum < target){
                    left++;
                }else {
                    right--;
                }
            }
        }

        return closest;
        
    }
}

File: 0016-3sum-closest/0016-3sum-closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {

    nums.sort((a,b) => a- b);

    let closest = nums[0] + nums[1] + nums[2];

    let n = nums.length;

    for(let i=0;i<n- 2;i++){
        let left = i+1;
        let right = n-1;
        
        while(left < right){

        let currentSum = nums[i] + nums[left] + nums[right];

        if(Math.abs(target - currentSum) < Math.abs(target - closest)){
            closest = currentSum;
        }

        if(currentSum < target){
            left++;
        }else if (currentSum > target){
            right--;
        }else {
            return currentSum;
        }
        }
    }
    return closest;
};